packages/dynamoose/lib/Table/index.ts

```diff
@@ -243,13 +243,21 @@ export class Table extends InternalPropertiesClass<TableInternalProperties> {
 					});
 				}, Promise.resolve());
 
-				await setupFlowPromise;
+				try {
+					await setupFlowPromise;
 
-				this.getInternalProperties(internalProperties).ready = true;
-				this.getInternalProperties(internalProperties).setupFlowRunning = false;
+					this.getInternalProperties(internalProperties).ready = true;
+					this.getInternalProperties(internalProperties).setupFlowRunning = false;
 
-				this.getInternalProperties(internalProperties).pendingTasks.forEach((task) => task());
-				this.getInternalProperties(internalProperties).pendingTasks = [];
+					this.getInternalProperties(internalProperties).pendingTasks.forEach((task) => task.resolve());
+					this.getInternalProperties(internalProperties).pendingTasks = [];
+				} catch (error) {
+					this.getInternalProperties(internalProperties).ready = false;
+					this.getInternalProperties(internalProperties).setupFlowRunning = false;
+
+					this.getInternalProperties(internalProperties).pendingTasks.forEach((task) => task.reject(error));
```

Comment:
I think this is the only line we need in the `catch` block, right?

Reply:
yes, this is the most necessary line in that block. I put the 2 lines above because I saw the logic and thought it might be necessary.

Reply:
@fishcharlie oh I forgot to run the test code.